src/ngx_http_lua_log.c

```diff
@@ -428,6 +428,30 @@ ngx_http_lua_ffi_errlog_get_sys_filter_level(ngx_http_request_t *r)
     return log_level;
 }
 
+
+int
+ngx_http_lua_ffi_raw_log(ngx_http_request_t *r, int level, u_char *s,
+    size_t s_len, u_char *err, size_t *errlen)
+{
+    ngx_log_t           *log;
+
+    if (level > NGX_LOG_DEBUG || level < NGX_LOG_STDERR) {
+        *errlen = ngx_snprintf(err, *errlen, "bad log level: %d", level) - err;
+        return NGX_ERROR;
+    }
+
+    if (r && r->connection && r->connection->log) {
+        log = r->connection->log;
+
+    } else {
+        log = ngx_cycle->log;
+    }
+
+    ngx_log_error((ngx_uint_t) level, log, 0, "[lua] %*s", s_len, s);
```

Comment:
Oh well, I actually didn't want it but thought it would be preferable, happy to remove it!